py7zr/py7zr.py

```diff
@@ -892,6 +911,35 @@ def write(self, file: Union[pathlib.Path, str], arcname: Optional[str] = None):
         folder = self.header.main_streams.unpackinfo.folders[-1]
         self.worker.archive(self.fp, self.files, folder, deref=self.dereference)
 
+    def writef(self, bio: BinaryIO, arcname: str):
+        if isinstance(bio, io.BytesIO):
+            size = bio.getbuffer().nbytes
+        elif isinstance(bio, io.BufferedIOBase):
+            size = bio.__sizeof__()
+        elif isinstance(bio, io.TextIOBase):
+            raise ValueError("Unsupported file object type: please open file with Binary mode.")
+        elif hasattr(bio, "read"):
+            # Unkown objet type but it has read() method; allow duck typing
+            pass
+        else:
+            raise ValueError("Wrong argument passed as BinaryIO.")
+        file_info = self._make_file_info_from_name(bio, size, arcname)
```

Comment:
It seems like `size` is not defined in the case bio has a `read` method but is not a known type.

Reply:
now checks whether object has `__sizeof__()` and usage of alternative methods, solved.